java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -451,33 +452,38 @@ public void accept(SymbolicValue symbolicValue, ConstraintsByDomain constraintPM
   }
 
   ProgramState resetFieldValues(ConstraintManager constraintManager, boolean resetOnlyStaticFields) {
-    final List<VariableTree> variableTrees = new ArrayList<>();
+    final List<Symbol> fields = new ArrayList<>();
     values.forEach((symbol, symbolicValue) -> {
       if (isField(symbol) && !symbol.isFinal() && (symbol.isStatic() || !resetOnlyStaticFields) ) {
-        VariableTree variable = ((Symbol.VariableSymbol) symbol).declaration();
-        if (variable != null) {
-          variableTrees.add(variable);
-        }
+        fields.add(symbol);
       }
     });
-    if (variableTrees.isEmpty()) {
+    if (fields.isEmpty()) {
       return this;
     }
     PMap<Symbol, SymbolicValue> newValues = values;
     PMap<SymbolicValue, Integer> newReferences = references;
-    for (VariableTree variableTree : variableTrees) {
-      Symbol symbol = variableTree.symbol();
-      SymbolicValue oldValue = newValues.get(symbol);
-      if (oldValue != null) {
+    for (Symbol field : fields) {
+      SymbolicValue oldValue = newValues.get(field);
+      if (field != null) {
         newReferences = decreaseReference(newReferences, oldValue);
       }
-      SymbolicValue newValue = constraintManager.createSymbolicValue(variableTree);
-      newValues = newValues.put(symbol, newValue);
+      Tree declarationOrFirstUsage = declarationOrFirstUsage(field);
```

Comment:
I think it is not necessary to search for a tree of a symbol. This was always `VariableTree` before which leads to call of `constraintManager.createDefaultSymbolicValue`. Now it is possible that this will be `IdentifierTree` , which can lead to some FPs when we accidentally trigger some logic in constraint manager like contrived example below

```
class B {
 Boolean TRUE;
}
class A extends B {

  void testUsage() {
    TRUE = Boolean.TRUE;
    TRUE = Boolean.FALSE;
    doSomething();
    if (TRUE == Boolean.FALSE) {  // FP condition always false

    }
  }
}
```
So I think it will be correct to make sure we are creating "fresh" symbolic value using `constraintManager.createDefaultSymbolicValue()`  with added benefit that we don't need the `Tree` for the symbol.